src/simple_thread_pool.cpp

```diff
@@ -0,0 +1,57 @@
+#include "print.hpp"
+#include "program_options.hpp"
+
+#include <boost/asio/post.hpp>
+#include <boost/asio/thread_pool.hpp>
+#include <boost/program_options.hpp>
+#include <boost/range/irange.hpp>
+
+#include <fmt/core.h>
+#include <thread>
+#include <chrono>
+
+namespace profitview
+{
+
+struct ProgramArgs
+{
+    int number_of_threads;
+
+    void addOptions(boost::program_options::options_description& options)
+    {
+        namespace po = boost::program_options;
+        // clang-format off 
+        options.add_options()
+            ("threads", po::value<int>(&number_of_threads)->required(), "Number of threads to create");
+        // clang-format on
+    }
+};
+
+}    // namespace profitview
+
+int main(int argc, char* argv[])
+{
+    using namespace profitview;
+    ProgramArgs options;
+    auto const result = parseProgramOptions(argc, argv, options);
+    if (result)
+        return result.value();
+
+    fmt_ns::print("Pool will have {} threads\n", options.number_of_threads);
```

Comment:
This will compile on Gcc/Clang but not Windows.  That is because the `print()` function is in the `print_ns` alias.  On Gcc and Clang this is the same as the `fmt_ns` alias because neither compiler has implemented either of these so they both point to `fmt`.  

But on Window VS has implemented `std::format` but not `std::print`.  So `print_ns` aliases `fmt` but the `fmt_ns` aliases `std`.  As such this would results in a compiler error as you would be asking for `std::print` not `fmt::print` 

Visual Studio 
```suggestion
    print_ns::print("Pool will have {} threads\n", options.number_of_threads);
```